adagrams/game.py

```diff
@@ -1,11 +1,175 @@
+import random
+from collections import Counter
+# letter_pool is a dict saving letters and their frequencies 
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+SCORES = {
+    'A': 1, 
+    'B': 3, 
+    'C': 3, 
+    'D': 2, 
+    'E': 1, 
+    'F': 4,
+    'G': 2, 
+    'H': 4,
+    'I': 1, 
+    'J': 8, 
+    'K': 5, 
+    'L': 1, 
+    'M': 3, 
+    'N': 1, 
+    'O': 1, 
+    'P': 3,
+    'Q': 10, 
+    'R': 1, 
+    'S': 1, 
+    'T': 1, 
+    'U': 1, 
+    'V': 4, 
+    'W': 4, 
+    'X': 8,
+    'Y': 4, 
+    'Z': 10
+}
+# I know you ask me to remove unnecessary comments, I'd prefer to keep them to see different ways.
+# def draw_letters():
+#     # Create an empty list to hold the drawn letters
+#     hand = []
+    
+#     # Create a dictionary to keep track of the count of each letter drawn
+#     letter_counts = {letter: 0 for letter in LETTER_POOL}
+    
+#     # Draw 10 letters, one at a time
+#     for i in range(10):
+#         # Randomly select a letter from the list of keys in the LETTER_POOL dictionary
+#         letter = random.choice(list(LETTER_POOL.keys()))
+        
+#         # If the letter count is less than its value in LETTER_POOL, add it to the hand
+#         if letter_counts[letter] < LETTER_POOL[letter]:
+#             hand.append(letter)
+#             letter_counts[letter] += 1
+#         # Otherwise, try again
+#         else:
+#             i -= 1
+    
+#     # Return the list of drawn letters
+#     return hand
```

Comment:
This one occasionally only returns a `hand` of 9 letters! See if you can figure out why.

To prevent this sort of issue, you can loop *while* the hand isn't big enough, e.g.:

```python
    while len(hand) < 10:
        letter = random.choice(list(LETTER_POOL.keys()))
        if letter_counts[letter] < LETTER_POOL[letter]:
            hand.append(letter)
            letter_counts[letter] += 1
```